Vi/add cities

remoderner opened the pull request:
add cities game

Commit:
Merge pull request #8 from ReforgedDream/development

Development

remoderner committed:
add cities game

d1z1k commented on src/main/java/cities/CitiesDB.java:
```diff
@@ -0,0 +1,71 @@
+    private Character lastChar;
+    private String botLastStep;
+
+    {
```
Это так конструктор можно записывать

remoderner replied:
Это не конструктор, а вызов метода при объявлении класса

remoderner committed:
- fix warnings in CitiesDB class
- add method StartGameCities to Cities class
- delete main method from Cities class

remoderner committed:
refactoring Cities and Cities DB classes
add CitiesChecks class

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,68 @@
+            }
+            citiesDB.setUsedCities(userInput); //Добавить последний ход в базу использованных городов
+            citiesChecks.setLastStep(userInput); //Запомнить последний ход
+            System.out.println("Успешно! Ход бота:");
```
Давай изменим ход бота на ход противника

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,68 @@
+    private CitiesDB citiesDB = new CitiesDB();
+    private CitiesChecks citiesChecks = new CitiesChecks();
+
+    public void StartGameCities() { //Запуск игры "Города"
```
Название метода с маленькой буквы

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,68 @@
+    private CitiesChecks citiesChecks = new CitiesChecks();
+
+    public void StartGameCities() { //Запуск игры "Города"
+        String title = "Начнем игру!";
```
это же final?

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,68 @@
+
+        for (; isGameContinue; ) {
+            cities.UserStep();
+            cities.BotStep();
```
Создавать объект cities нет необходимости, можно просто вызывать метода

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,68 @@
+            cities.BotStep();
+        }
+    }
+    private void UserStep() { //Ход пользователя
```
с маленькой буквы

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,68 @@
+                break; //Выход из цикла, для продолжения игры
+            } else if (i++ == citiesDB.getСitiesList().size() - 1) {
+                System.out.println("Бот не может назвать город! Вы победили!");
+                isGameContinue = false;
```
inGameContinue можно заменить на то что метод с ходом бота или ходом человека возвращает тру или фолс и от этого делать проверку, нужен следующий ход или это тупик

remoderner committed:
fix bug with empty readline
fix print when user end step

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,69 @@
+                System.out.println(e.toString());
+            }
+            if (userInput.trim().length() == 0) {
+                continue;
```
А почему здесь нету диагностики?

d1z1k commented on src/main/java/cities/Cities.java:
```diff
@@ -0,0 +1,69 @@
+            cities.BotStep();
+        }
+    }
+    private void UserStep() { //Ход пользователя
```
отсутствует перенос строки

d1z1k replied:
между }  и объявлением метода

remoderner committed:
refactoring
add filereader to addCities method
add CitiesOfRussia.txt

remoderner committed:
refactoring
add checkValidLastChar method to CitiesChecks class
improvement getLastChar in CitiesChecks class
fix infinitive cycle from userStep method in Cities class

remoderner committed:
improvement startGameCities method in Cities class

remoderner committed:
add CitiesOfTheWorld.txt

remoderner committed:
refactoring
